### ISSUE TITLE
saltydb missing inner join used in default template

### ISSUE THREAD

**Issue by lheberlie** [jsdoc/jsdoc#2103]
Taffy innerJoin Extension (OCD edition) missing from `@jsdoc/salty`

TLDR;
When upgrading from version 3 to version 4, the default template is missing some functionality to `attachModules` since version 3 uses taffydb inner join and version 4 uses saltydb which is missing the inner join.

Request:

Please provide a solution for the existing query
```js
attachModuleSymbols( find({ longname: {left: 'module:'} }), members.modules );
```
or provide a solution or suggestion to update the `attachModuleSymbols` function to fix the default template.

Thank you.

The [default template](  https://github.com/jsdoc/jsdoc/blob/releases/4.0/templates/default/publish.js#L603`) uses a [left join]()  from the original [taffydb](https://github.com/typicaljoe/taffydb/blob/master/taffy.js#L880C9-L880C48) to send data to the [`attachModules`](https://github.com/jsdoc/jsdoc/blob/releases/4.0/templates/default/publish.js#L274) function.
This is missing in saltyDB and I'm wondering if you have a solution for the default template or what's the level of effort to bring this into saltyDB so the default template still works properly.

### Input code

```js
attachModuleSymbols( find({ longname: {left: 'module:'} }), members.modules );
```
https://github.com/jsdoc/jsdoc/blob/0b6193f185b66ca7502b145acdc66f4579c63e8d/templates/default/publish.js#L603C5-L603C83

### Expected behavior

Version: `3.6.10`

#### taffydb results:
![image](https://github.com/jsdoc/jsdoc/assets/1824760/e3c2dd04-643e-431d-b2b8-942345c5824f)

#### code execution
```js
helper.find(data, { kind: ["class", "function"], longname: { left: "module:" } })
‌
Array(7262) [Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, Doclet, ...]
```

### Current behavior

Version: `4.0.2`

#### saltydb data sample since the results are empty:
![image](https://github.com/jsdoc/jsdoc/assets/1824760/47221ebe-8f0f-4ab1-809f-b1320532e42c)

#### code execution:
```js
helper.find(data, { kind: ["class", "function"], longname: { left: "module:" } })
‌
Array(0) []
```

### Your environment

| Software         | Version |
| ---------------- | ------- |
| JSDoc            | `4.0.2` |
| Node.js          | `v20.10.0` |
| npm              | `10.2.3` |
| Operating system | `macOS 14.2.1` |

**Comment by lheberlie** [comment]
@hegemonic - curious if you have seen this issue yet?

**Comment by lheberlie** [comment]
@hegemonic - hoping this isn't lost in the email abyss.

**Comment by hegemonic** [comment]
Fixed in `@jsdoc/salty@0.2.8`. If you reinstall JSDoc, you should pick up the new version automatically.

**Comment by lheberlie** [comment]
Thank you @hegemonic
